src/autoscaler/api/policyvalidator/policy_validator.go

```diff
@@ -113,67 +126,113 @@ func (pv *PolicyValidator) validateAttributes(policy *models.ScalingPolicy, resu
+				result.AddError(err, errDetails)
+			}
+		case "cpu":
+			if scalingRule.Threshold <= 0 || scalingRule.Threshold > 100 {
+				formatString := "scaling_rules[{{.scalingRuleIndex}}].threshold for metric_type cpu should be greater than 0 and less than equal to 100"
+				err := newPolicyValidationError(currentContext, formatString, errDetails)
+				result.AddError(err, errDetails)
+			}
+		default:
+		}
+	}
+}
+
 func (pv *PolicyValidator) validateRecurringSchedules(policy *models.ScalingPolicy, schedulesContext *gojsonschema.JsonContext, result *gojsonschema.Result) {
 	recurringScheduleContext := gojsonschema.NewJsonContext("recurring_schedule", schedulesContext)
-	for i, recSched := range policy.Schedules.RecurringSchedules {
+	for scheduleIndex, recSched := range policy.Schedules.RecurringSchedules {
 		if recSched.ScheduledInstanceMin >= recSched.ScheduledInstanceMax {
-			instanceMinContext := gojsonschema.NewJsonContext(fmt.Sprintf("%d.instance_min_count", i), recurringScheduleContext)
+			instanceMinContext := gojsonschema.NewJsonContext(fmt.Sprintf("%d.instance_min_count", scheduleIndex), recurringScheduleContext)
 			errDetails := gojsonschema.ErrorDetails{
+				"scheduleIndex":      scheduleIndex,
 				"instance_min_count": recSched.ScheduledInstanceMin,
 				"instance_max_count": recSched.ScheduledInstanceMin,
 			}
-			formatString := "instance_min_count {{.instance_min_count}} is higher or equal to instance_max_count {{.instance_max_count}}"
+			formatString := "recurring_schedule[{{.scheduleIndex}}].instance_min_count {{.instance_min_count}} is higher or equal to recurring_schedule[{{.scheduleIndex}}].instance_max_count {{.instance_max_count}}"
 			err := newPolicyValidationError(instanceMinContext, formatString, errDetails)
 			result.AddError(err, errDetails)
 		}
 
-		if recSched.ScheduledInstanceInit > recSched.ScheduledInstanceMax {
-			initialInstanceMinContext := gojsonschema.NewJsonContext(fmt.Sprintf("%d.initial_min_instance_count", i), recurringScheduleContext)
+		if (recSched.ScheduledInstanceInit != 0) && (recSched.ScheduledInstanceInit < recSched.ScheduledInstanceMin) {
+			initialInstanceMinContext := gojsonschema.NewJsonContext(fmt.Sprintf("%d.initial_min_instance_count", scheduleIndex), recurringScheduleContext)
 			errDetails := gojsonschema.ErrorDetails{
+				"scheduleIndex":              scheduleIndex,
 				"initial_min_instance_count": recSched.ScheduledInstanceInit,
-				"instance_max_count":         recSched.ScheduledInstanceMin,
+				"instance_min_count":         recSched.ScheduledInstanceMin,
 			}
-			formatString := "initial_min_instance_count {{.initial_min_instance_count}} is greater than instance_max_count {{.instance_max_count}}"
+			formatString := "recurring_schedule[{{.scheduleIndex}}].initial_min_instance_count {{.initial_min_instance_count}} is smaller than recurring_schedule[{{.scheduleIndex}}].instance_min_count {{.instance_min_count}}"
```

Comment:
it is good to check initial_min >= min,  but we still need the original check of initial_min < max as well. please not delete the original one ..

Reply:
done.